PORTING-ADDONS-TO-V2.md

```diff
@@ -0,0 +1,264 @@
+
+## Monorepo Organization
+
+Traditionally, an Ember addon is a single NPM package that combines both the actual addon code _and_ a "dummy" app for hosting tests and docs. This was [problematic for several reasons](https://github.com/ember-cli/rfcs/issues/119). V2 addons instead require clean separation between addon and app, so you're going to be working with more than one distinct NPM package: one for the addon, one for the test-app, and optionally one for a documentation site.
+
+Our recommended way to manage these multiple packages is using a monorepo, via Yarn Workspaces or NPM Workspaces. The example in this guide assumes a Yarn Workspaces monorepo.
+
+## Part 1: Separate Addon from Dummy App
+
+In this part of the guide, our goal is to separate our existing V1 addon from its "dummy" app and rename the dummy app to "test-app". At the end of this part, you will still have a V1 addon but it will be independent of its test-app, making it much easier to convert to V2 format in a subsequent Part.
+
+For a complete example of a PR that performed these steps on a real addon, see https://github.com/ember-cli/ember-page-title/pull/227.
+
+The steps:
+
+1. Delete yarn.lock.
+
+1. At the top-level of your repo, make new directories named `test-app` and `_addon`
+
+1. Move these files and directories into `_addon`
+
+   - addon
+   - addon-test-support
+   - app
+   - blueprints
+   - config/environment.js (moves to `_addon/config/environment.js`)
+   - index.js
+
+1. Now you can rename `_addon` to `addon` without a name collision.
+
+   - yes, this means you will have an `addon/addon` directory. This looks silly but it will go away when we finish porting the addon to v2.
+
+1. These things stay at the top level:
+
+   - .git
+   - .github
+   - changelog, code of conduct, contributing, license, and readme
+
+   Move **everything else** into `test-app`
+
+1. Move everything under `test-app/tests/dummy` to directly under `test-app` instead.
+
+   - for example, `test-app/tests/dummy/app` becomes `test-app/app`
+   - you will be merging config directories, because both `test-app/config` and `test-app/tests/dummy/config` will exist at the start of this step. They shouldn't have any file collisions though, because you already moved the one colliding file (`environment.js`) to `addon/config/environment.js` in a previous step.
+
+1. Make a new top-level package.json for our new monorepo:
+
+   ```
+   {
+     private: true,
+     workspaces: [
+       "addon",
+       "test-app"
+     ]
+   }
+   ```
+
+1. Make a new top-level .gitignore:
+
+   ```
+   # you definitely want this:
+   node_modules
+
+   # and you can put in anything else that tends to accumulate in your environment:
+   yarn-error.log
+   .DS_Store
```

Comment:
we may mention `.github` as it's often overlooked

Reply:
Is there stuff in `.github` that should be gitignored?

Reply:
ah sorry, this is `.gitignore` and I read `.npmignore` 🙈